src/Chainweb/Mempool/InMem.hs

```diff
@@ -228,11 +230,23 @@ insertInMem cfg lock runCheck txs0 = do
                         else return
     preInsertFilterReal txs = do
         let chk = _inmemPreInsertCheck cfg
-        let out1 = V.map sizeOK txs
-        out2 <- chk txs
-        let oks = V.zipWith (&&) out1 out2
+
```

Comment:
Most of the new code in this function should probably just live inside `_inmemPreInsertCheck`

Reply:
I cleaned up the code with this comment. The checks have been moved in a particular way. You'll see what I mean when you see the changes.